### PR TITLE
JSON logging [TW-150] [CSL-1199]

### PR DESCRIPTION
**Pull request by brunjlar**
- moved JSON logging from cardano-sl to time-warp-nt
- added all JSON log events from the benchmark branch

### CODE REVIEW

**Comment by gromakovsky** [comment]
Do we need it for cardano-sl-0.5?

**Comment by brunjlar** [comment]
Yes - that's why I went through a lot of trouble to shift it from `master` to `cardano-sl-0.5-rc`.

**Comment by kantp** [comment]
I've had a look at the CI failures, and they seem to be unrelated to the contents of the pull request:
We have

```util-scripts/lint.sh: No such file or directory```

and

```
[00:05:27] 'scripts\appveyor-retry' is not recognized as an internal or external command,
[00:05:27] operable program or batch file.
```
@domenkozar Do these error messages tell you anything?

**Comment by volhovM** [comment]
@kantp i think the first error is related to script reorganisation we did several days ago:
https://github.com/input-output-hk/cardano-sl/pull/945
The new path for lint is `scripts/haskell/lint.sh`.

**Comment by kantp** [comment]
Ah, thanks @volhovM. In that case, merging `cardano-sl-0.5-rc` again should fix this.

File: infra/Pos/Communication/Types/Protocol.hs [hunk 1]
```diff
@@ -77,6 +81,22 @@ data Conversation m t where
         => (N.ConversationActions snd rcv m -> m t)
         -> Conversation m t
 
+newtype PeerId = PeerId ByteString
```
Comment (gromakovsky) [inline infra/Pos/Communication/Types/Protocol.hs]: What is `PeerId`? Is it needed? Currently it's not used anywhere. To be more precise, it's used in `TransactionProvenance` which is used in `MemPoolModifyReason`, but it's not used at all.

**Comment by gromakovsky** [comment]
Now you should merge 5b922a5069533d3f18cae85c4a3113020642adc4 to `master`.

**Comment by gromakovsky** [comment]
@brunjlar @kantp or someone else. Please merge 5b922a5069533d3f18cae85c4a3113020642adc4 to master as soon as you find time for it. It's very important to not delay this merge.

**Pull request merged**